sdk/storage/src/core/bearer_token_client.rs

```diff
@@ -113,7 +114,7 @@ impl<'a> Client for BearerTokenClient<'a> {
         method: &Method,
         http_header_adder: &dyn Fn(Builder) -> Builder,
         request_body: Option<&[u8]>,
-    ) -> Result<hyper::client::ResponseFuture, AzureError> {
+    ) -> Result<(url::Url, hyper::client::ResponseFuture), AzureError> {
```

Comment:
I'm not sure why this returns a `Url`, can you explain?

Reply:
It's needed to construct the `ContinuationToken` in the response. Continuation tokens require:

1. The exact URL called the first time.
2. The token ID (in the case of table storage is partition_key and row_key). 

I also do not like the solution of polluting the return with a tuple. Do you think it would be better to create a specific struct instead?

Reply:
Yes, I believe a struct would be much more appropriate here.

Reply:
Done (tuple replaced with PerformRequestResponse struct)